src/jdk.attach/aix/classes/sun/tools/attach/VirtualMachineImpl.java

```diff
@@ -57,51 +56,46 @@ public class VirtualMachineImpl extends HotSpotVirtualMachine {
         super(provider, vmid);
 
         // This provider only understands pids
-        int pid;
-        try {
-            pid = Integer.parseInt(vmid);
-            if (pid < 1) {
-                throw new NumberFormatException();
-            }
-        } catch (NumberFormatException x) {
-            throw new AttachNotSupportedException("Invalid process identifier: " + vmid);
+        int pid = Integer.parseInt(vmid);
```

Comment:
This will now throw `NumberFormatException` instead of the expected `AttachNotSupportedException`.

Reply:
The same check is performed in its superclass HotSpotVirtualMachine. So we can be sure that it will not throw NumberFormatException here

https://github.com/openjdk/jdk/blob/38cfc591725de478879266584280562f0ba4b42f/src/jdk.attach/share/classes/sun/tools/attach/HotSpotVirtualMachine.java#L69-L74

Reply:
In that case the `int pid` determined by the superclass should be made available to the subclasses so they can use it and not need to re-parse the `vmid`.

Reply:
It is well within scope of improving code sharing in this arera, but I'm okay with leaving that for a next step.